### PR TITLE
[nereids](map_agg)support map_agg in nereids

### PR DESCRIPTION
**Pull request by LemonLiTree**
support map_agg in nereids

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/MapAgg.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+/**
+ * AggregateFunction 'map_agg'. This class is generated by GenerateFunction.
+ */
+public class MapAgg extends NullableAggregateFunction
+    implements BinaryExpression, ExplicitlyCastableSignature {
+
+    public static final List<FunctionSignature> SIGNATURES = ImmutableList.of(
+        FunctionSignature.ret(MapType.of(new FollowToAnyDataType(0), new FollowToAnyDataType(1)))
+            .args(AnyDataType.INSTANCE_WITHOUT_INDEX, AnyDataType.INSTANCE_WITHOUT_INDEX)
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/MapAgg.java:42]:
```suggestion
        FunctionSignature.ret(MapType.of(new FollowToAnyDataType(0), new FollowToAnyDataType(1)))
            .args(new AnyDataType(0), new AnyDataType(1))
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/MapAgg.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+    /**
+     * constructor with 2 arguments.
+     */
+    public MapAgg(boolean distinct, Expression arg0, Expression arg1) {
+        this(distinct, false, arg0, arg1);
+    }
+
+    private MapAgg(boolean distinct, boolean alwaysNullable, Expression arg0, Expression arg1) {
+        super("map_agg", distinct, false, arg0, arg1);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/MapAgg.java:60]:
```suggestion
        super("map_agg", distinct, alwaysNullable, arg0, arg1);
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/MapAgg.java (hunk 1 above)
Comment (LemonLiTree) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/agg/MapAgg.java]: done

**Comment by LemonLiTree** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

**Comment by morrySnow** [comment]
run buildall

**Review by keanji-x — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**